### PR TITLE
Library refactoring

### PR DESCRIPTION
**Pull request by dittos**

### CODE REVIEW

File: frontend/js/layouts.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
+import {Link} from 'nuri';
```
Comment (dittos) [inline frontend/js/layouts.js]: asdf

File: frontend/js/routes/Chart.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import {Link} from 'nuri';
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: Hi

File: frontend/js/routes/Chart.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
...
 import {App} from '../layouts';
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: asdf

Comment (dittos) [inline frontend/js/routes/Chart.js]: ggg

File: frontend/js/routes/Chart.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
...
 import * as Layout from '../ui/Layout';
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: adaffdsaafds

File: frontend/js/routes/Index.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ import WeeklyChart from '../ui/WeeklyChart';
 import PostComment from '../ui/PostComment';
 import LoadMore from '../ui/LoadMore';
 import Styles from '../../less/index.less';
+// TODO: css module
```
Comment (dittos) [inline frontend/js/routes/Index.js]: ya

File: frontend/js/routes/Index.js [hunk 6]
```diff
@@ -9,6 +9,7 @@ import WeeklyChart from '../ui/WeeklyChart';
...
```
Comment (dittos) [inline frontend/js/routes/Index.js]: asdfaf

File: frontend/js/ui/WeeklyChart.js [hunk 7]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import {Link} from 'nuri';
 import {getWorkURL} from '../util';
 import Styles from './WeeklyChart.less';
+// TODO: css module
```
Comment (dittos) [inline frontend/js/ui/WeeklyChart.js]: wut

File: frontend/js/routes/AddRecord.js [hunk 8]
```diff
@@ -1,11 +1,8 @@
...
-var {withRouter} = require('react-router');
-var {connect} = require('react-redux');
-var RecordActions = require('../store/RecordActions');
-var CategoryStore = require('../store/CategoryStore');
-var Typeahead = require('./Typeahead');
+import React from 'react';
+import {createRecord} from '../API';
+import {User} from '../layouts';
+import Typeahead from '../ui/Typeahead';
+// TODO: css module
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: ho

File: frontend/js/routes/Post.js [hunk 9]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import * as WorkViews from '../ui/WorkViews';
 import {getStatusDisplay} from '../util';
```
Comment (dittos) [inline frontend/js/routes/Post.js]: xyz

File: frontend/js/routes/AddRecord.js [hunk 10]
```diff
@@ -1,11 +1,8 @@
-var $ = require('jquery');
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: xxx

File: frontend/js/routes/Chart.js [hunk 11]
```diff
@@ -63,7 +64,6 @@ class Chart extends React.Component {
     render() {
         const {
             chart,
-            chartType,
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: asdf

File: frontend/js/routes/AddRecord.js [hunk 12]
```diff
@@ -1,11 +1,8 @@
...
-var React = require('react');
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: gg

File: frontend/js/routes/ManageCategory.js [hunk 13]
```diff
@@ -0,0 +1,184 @@
+/* global confirm */
+import React from 'react';
+import cx from 'classnames';
```
Comment (dittos) [inline frontend/js/routes/ManageCategory.js]: x

File: frontend/js/routes/AddRecord.js [hunk 14]
```diff
@@ -1,11 +1,8 @@
...
-var ReactDOM = require('react-dom');
```
Comment (dittos-test) [inline frontend/js/routes/AddRecord.js]: what

File: frontend/js/layouts.js [hunk 15]
```diff
@@ -11,3 +13,25 @@ export function App(Component) {
         <Component {...props} />
     </div>;
 }
+
+// TODO: css module
+export function User(Component) {
```
Comment (dittos-test) [inline frontend/js/layouts.js]: 이렇게

File: frontend/js/layouts.js [hunk 16]
```diff
@@ -11,3 +13,25 @@ export function App(Component) {
...
+    return App((props) => {
```
Comment (dittos-test) [inline frontend/js/layouts.js]: 마지막 리뷰 댓글만 안나온다.

File: frontend/js/routes/AddRecord.js [hunk 17]
```diff
@@ -1,11 +1,8 @@
...
-var {withRouter} = require('react-router');
```
Comment (dittos-test) [inline frontend/js/routes/AddRecord.js]: ㅎㅎ

File: frontend/js/routes/AddRecord.js [hunk 18]
```diff
@@ -31,7 +28,7 @@ var CategorySelect = React.createClass({
 var AddRecord = React.createClass({
     getInitialState() {
         return {
-            selectedCategoryId: 0,
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: asdf

File: frontend/js/routes/AddRecord.js [hunk 19]
```diff
@@ -1,11 +1,8 @@
-var $ = require('jquery');
-var React = require('react');
-var ReactDOM = require('react-dom');
-var {withRouter} = require('react-router');
-var {connect} = require('react-redux');
-var RecordActions = require('../store/RecordActions');
-var CategoryStore = require('../store/CategoryStore');
-var Typeahead = require('./Typeahead');
+import React from 'react';
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: ggg

File: frontend/js/routes/AddRecord.js [hunk 20]
```diff
@@ -1,11 +1,8 @@
...
+import {createRecord} from '../API';
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: ggggggg

File: frontend/js/routes/AddRecord.js [hunk 21]
```diff
@@ -62,8 +59,8 @@ var AddRecord = React.createClass({
                 <tr>
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: asdf

File: frontend/js/routes/AddRecord.js (hunk 17 above)
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: a

File: frontend/js/routes/AddRecord.js [hunk 22]
```diff
@@ -1,11 +1,8 @@
-var $ = require('jquery');
-var React = require('react');
-var ReactDOM = require('react-dom');
-var {withRouter} = require('react-router');
-var {connect} = require('react-redux');
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: yeah

File: frontend/js/routes/AddRecord.js [hunk 23]
```diff
@@ -1,11 +1,8 @@
...
-var RecordActions = require('../store/RecordActions');
```
Comment (dittos) [inline frontend/js/routes/AddRecord.js]: test

File: frontend/js/routes/Chart.js [hunk 24]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import {Link} from 'nuri';
 import {App} from '../layouts';
 import * as Layout from '../ui/Layout';
+// TODO: css module
 
 class Header extends React.Component {
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: yesssss

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/routes/Chart.js [hunk 25]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
...
     render() {
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: asdf

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/routes/Chart.js [hunk 26]
```diff
@@ -63,7 +64,6 @@ class Chart extends React.Component {
...
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: ggggg

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/routes/Chart.js [hunk 27]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import {Link} from 'nuri';
 import {App} from '../layouts';
 import * as Layout from '../ui/Layout';
+// TODO: css module
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: ok

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/routes/Chart.js (hunk 4 above)
Comment (dittos) [inline frontend/js/routes/Chart.js]: xxx

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/routes/Chart.js (hunk 2 above)
Comment (dittos) [inline frontend/js/routes/Chart.js]: start

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/routes/Chart.js (hunk 3 above)
Comment (dittos) [inline frontend/js/routes/Chart.js]: 2

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

Comment (dittos) [inline frontend/js/routes/Chart.js]: 3

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/routes/Chart.js (hunk 2 above)
Comment (dittos) [inline frontend/js/routes/Chart.js]: hhh

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/routes/AddRecord.js [hunk 28]
```diff
@@ -1,11 +1,8 @@
...
-var CategoryStore = require('../store/CategoryStore');
```
Comment (dittos-test) [inline frontend/js/routes/AddRecord.js]: ho

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FAddRecord.js)_</sub>

File: frontend/js/routes/AddRecord.js [hunk 29]
```diff
@@ -1,11 +1,8 @@
...
-var Typeahead = require('./Typeahead');
+import React from 'react';
+import {createRecord} from '../API';
+import {User} from '../layouts';
+import Typeahead from '../ui/Typeahead';
```
Comment (dittos-test) [inline frontend/js/routes/AddRecord.js]: asdf

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FAddRecord.js)_</sub>

File: frontend/js/routes/AddRecord.js [hunk 30]
```diff
@@ -1,11 +1,8 @@
...
+// TODO: css module
```
Comment (dittos-test) [inline frontend/js/routes/AddRecord.js]: wtwetwtwt

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FAddRecord.js)_</sub>

File: frontend/js/routes/Index.js [hunk 31]
```diff
@@ -9,6 +9,7 @@ import WeeklyChart from '../ui/WeeklyChart';
 import PostComment from '../ui/PostComment';
```
Comment (dittos) [inline frontend/js/routes/Index.js]: hey

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FIndex.js)_</sub>

File: frontend/js/routes/Index.js [hunk 32]
```diff
@@ -9,6 +9,7 @@ import WeeklyChart from '../ui/WeeklyChart';
...
 import LoadMore from '../ui/LoadMore';
```
Comment (dittos) [inline frontend/js/routes/Index.js]: asdf

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FIndex.js)_</sub>

File: frontend/js/routes/Index.js [hunk 33]
```diff
@@ -9,6 +9,7 @@ import WeeklyChart from '../ui/WeeklyChart';
...
 import Styles from '../../less/index.less';
```
Comment (dittos) [inline frontend/js/routes/Index.js]: gg

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FIndex.js)_</sub>

File: frontend/js/routes/Index.js (hunk 5 above)
Comment (dittos) [inline frontend/js/routes/Index.js]: ggg

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FIndex.js)_</sub>

File: frontend/js/ui/PostComposer.js [hunk 34]
```diff
@@ -2,26 +2,31 @@ var React = require('react');
 var LinkedStateMixin = require('react-addons-linked-state-mixin');
 var StatusInput = require('./StatusInput');
 var util = require('../util');
-var ExternalServiceActions = require('../store/ExternalServiceActions');
+var LocalStorage = require('../LocalStorage');
```
Comment (dittos) [inline frontend/js/ui/PostComposer.js]: 안녕하세요.

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Fui%2FPostComposer.js)_</sub>

File: frontend/js/routes/Chart.js [hunk 35]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
...
```
Comment (dittos) [inline frontend/js/routes/Chart.js]: 힝 입니다

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

Comment (dittos) [inline frontend/js/routes/Chart.js]: 힝 이다

<sub>_commented via [Diff Monster](https://diff.sapzil.org/#/dittos/animeta/pull/58?path=frontend%2Fjs%2Froutes%2FChart.js)_</sub>

File: frontend/js/ui/PostComposer.less [hunk 36]
```diff
@@ -54,3 +54,12 @@
         font-size: 14px;
     }
 }
+
+:global(.plus-one) {
```
Comment (dittos) [inline frontend/js/ui/PostComposer.less]: 삭제 (StatusInput.less에 있음)

**Pull request closed without merging**